indirect.h

```diff
@@ -66,15 +66,18 @@ class indirect {
   using pointer = typename allocator_traits::pointer;
   using const_pointer = typename allocator_traits::const_pointer;
 
+  template <typename U = T>
   constexpr indirect()
-    requires std::is_default_constructible_v<A>
+    requires(std::is_default_constructible_v<A> &&
+             std::is_default_constructible_v<U>)
```

Comment:
This delays the evaluation of the constraint until this function is instantiated so it works with incomplete types?

Reply:
That’s my understanding. I’m not sure if that means we can graduate our _Mandates_ to a _Constraints_

Reply:
@lewissbaker does this PR mean we can/should change wording too?

Reply:
Yes. I think the wording would need to be updated to use Constraints.
The signature would probably also need to be updated to:
```
template<typename U = T>
constexpr indirect();
```
Constraints: `is_default_constructible_v<U>` is `true` and `is_default_constructible_v<Alloc>` is `true`.